transformers/senml/transformer.go

```diff
@@ -20,20 +27,19 @@ var formats = map[string]senml.Format{
 	CBOR: senml.CBOR,
 }
 
-type transformer struct{}
+type transformer struct {
+	format senml.Format
+}
 
 // New returns transformer service implementation for SenML messages.
-func New() transformers.Transformer {
-	return transformer{}
+func New(contentType string) transformers.Transformer {
+	return transformer{
+		format: formats[contentType],
+	}
```

Comment:
It makes sense to have a default format (also, please rename `contentType` to `contentFormat`):
```go
format, ok := formats[contentFormat]
if !ok {
    format = formats[JSON]
}

return transformer {
    format: format,
} 
```